airbyte-integrations/connectors/source-facebook-marketing/source_facebook_marketing/client/api.py

```diff
@@ -66,25 +72,79 @@ def state(self):
     def state(self, value):
         self._state = pendulum.parse(value[self.state_pk])
 
-    def __init__(self, *args, **kwargs):
+    def __init__(self, *args, include_deleted=False, **kwargs):
         super().__init__(*args, **kwargs)
+        self._include_deleted = include_deleted
         self._state = None
 
+    def _build_params(self, params: Mapping[str, Any] = None) -> Iterator:
+        """Build complete params for request"""
+        result_params = super()._build_params(params)
+        for filters in self._build_filters():
+            result_params.setdefault("filtering", []).extend(filters)
+
+            yield result_params
+
+    def _build_filters(self) -> Iterator[List[Mapping[str, Any]]]:
+        """Build complete filters"""
+        for delete_filter in self._delete_filters():
+            filters = []
+            state_filter = self._state_filter()
+            if state_filter:
+                filters.append(state_filter)
+            if delete_filter:
+                filters.append(delete_filter)
+
+            yield filters
+
     def _state_filter(self):
         """Additional filters associated with state if any set"""
         if self._state:
             return {
-                "filtering": [
-                    {
-                        "field": f"{self.entity_prefix}.{self.state_pk}",
-                        "operator": "GREATER_THAN",
-                        "value": self._state.int_timestamp,
-                    },
-                ],
+                "field": f"{self.entity_prefix}.{self.state_pk}",
+                "operator": "GREATER_THAN",
+                "value": self._state.int_timestamp,
             }
 
         return {}
 
+    def _delete_filters(self):
+        """ We split single request into multiple requests with few delivery_info values"""
```

Comment:
why?

Reply:
this was part of singer tap implementation, and I believe this is because querying many types of states at the same time could cause a slow response from the API

Reply:
could you add this context into the comment?